### DISCUSSION TITLE
Need help deploying sunbird in kubernetes

### DISCUSSION THREAD

**Discussion by sathyap0104** [Sunbird-Ed/Community#73] [Installation] [answered]
Hi team,
Greetings of the day,
   We are trying to deploy sunbird in the kubernetes environment. so, in the process we saw
1. Dockerfiles and to run them we need dist folder so we tried running those steps locally but we are stuck at environmentVariablesHelper.js.  
2. We also saw helm-charts, but not sure which chart goes first. and seems like they are interrelated and tried to deploy them but they are throwing null pointer errors. 
`Error: INSTALLATION FAILED: template: adminutils/templates/hpa.yaml:1:14: executing "adminutils/templates/hpa.yaml" at <.Values.autoscaling.enabled>: nil pointer evaluating interface {}.enabled`
3. saw the document from sunbird.org and tried to run on azure platform. But not sure which jenkins job to run first. 
Hoping for some support from your side. Maybe we are making some mistakes or not on the right path. Do we have any way to deploy it in local kubernetes.

**Comment by beepdot** [comment]
Please see all pages under Server Installation. They are in the order which you should follow.
http://docs.sunbird.org/latest/developer-docs/server-installation/

**Reply by sathyap0104** [reply]
HI, @keshavprasadms Thank you for the replay. Followed the same doc but still facing issues. While running Jenkins jobs only a few jobs are getting successful. Most of them are failing. And we didn't get the slave when we run that jenkins-setup.sh file too. And another query Do this jobs create images? if does where they will be built or pushed to?

**Reply by daveanvit** [reply, marked as answer]
1.  In jenkins.yaml file there are tags **build-slave ops-slave** check node configuration in manage Jenkins section, did those tags got applied or not. Also  check **Configuration as Code Plugin** is installed or not. **build-slave ops-slave** tags and other environment variables are applied from that plugin.
Note: Configuration as Code Plugin is not available in Jenkins version **2.277.4** you need to update Jenkis to install it.

2. Yes, this jobs do create images, images got pushed to dockerhub repo which we have configured in **jenkins.yaml** and **ansible ** configuration files.

**Reply by beepdot** [reply]
Please try the input as suggested by @daveanvit and let us know.

**Reply by sathyap0104** [reply]
Sure @keshavprasadms Thank you. Will update you after trying it.

**Comment by sharathkashyap** [comment]
@sathyap0104 Can you please confirm if this is working.

**Reply by sharathkashyap** [reply]
@sathyap0104 I may have to close the thread due to inactivity. Kindly raise a seperate thread if the issue is not resolved yet.
